wcivf/apps/elections/models.py

```diff
@@ -229,6 +254,27 @@ def nice_territory(self):
 
         return self.territory
 
+    @property
+    def division_description(self):
+        """
+        Return a string to describe the division
+        """
+        mapping = {
+            choice[0]: choice[1] for choice in self.DIVISION_TYPE_CHOICES
+        }
+        return mapping.get(self.division_type, "")
+
+    @property
+    def division_suffix(self):
+        """
+        Returns last word of the division_description
+        """
+        return self.division_description.split(" ")[-1].lower()
```

Comment:
I can see this causing problems in future - one of the hard things about this system of taking strings from EE and using them elsewhere is it's hard to remember how and where we're actually using them (this applies in many places, not just this PR!).

Relying on a particular format might cause problems in the future - especially as we might change EE without checking here.

I think this means a couple of things:

1. Finding a way to store how strings from EE are used, and writing that in the EE docs, so next time we add something we don't have to guess how to use it

2. Sort of as a guard against the above, I think we should store 3 bits of information: ID, description, suffix, maybe as a named tuple. We can grab the description as you're doing in `division_description, but then change `division_suffix` to return `division_description.suffix` rather than splitting the description.

Reply:
Not sure I completely follow - `division_description` is not something that is being stored from EE (for the reason you outlined - in the issue  `division_subtype` [is mentioned as something that should not be relied upon being present all of the time](https://github.com/DemocracyClub/WhoCanIVoteFor/issues/608#issuecomment-763849830)). 

So all I am storing from EE at the moment is the `division_type` which is a 3 letter code. And then I have hardcoded the descriptions for each code using the descriptions in issue #608. Therefore any change in EE wouldn't affect anything about how we determine the description for a `division_type`. Unless a new `division_type` is added in EE, in which case we would need to update the codebase here.

So as far as `division_suffix` goes I can then be confident that what when I do the `split`, assuming we have a valid `division_type` the format will match the way I expect (e.g. final word being a descriptor like "ward") because it is set here and not in EE. And if we don't have a `division_type` code (e.g. it is a Mayoral Election) then it will always return an empty string rather than error. I was then going to update this to account for those sort of edge cases somehow.

That being said, happy to implement the changes you have suggested, but I think it would mean capturing and storing some extra bits from EE (e.g. the description)? So perhaps best to have a quick run through to check we are on same page.

Reply:
Sorry, my mistake - as long as we're parsing text that's checked in to this repo I think we're good :+1: